src/scanner.rs

```diff
@@ -463,9 +463,13 @@ impl<T: Iterator<Item = char>> Scanner<T> {
             return;
         }
         for _ in 0..(count - self.buffer.len()) {
-            self.buffer
+            if self
+                .buffer
                 .push_back(self.rdr.next().unwrap_or('\0'))
-                .unwrap();
+                .is_err()
+            {
+                break;
+            }
```

Comment:
I don't think this change is the right thing to do. The `buffer` errors when we try to push past its capacity (since it is fixed and cannot reallocate). This is something that, if we hit, is because `count > self.buffer.len()`, which is a logic error in the scanner that should be fixed not in `lookahead` but on its call site.

The `unwrap` here should never be hit. Breaking out of the loop might trigger a plethora of panics in the scanner, since the code assumes that, after calling `lookahead(n)`, it is safe to access `buffer[0..n-1]`.

In my opinion, the best course of action here would be to rather `assert(count <= self.buffer.len())`, ensuring we never call `lookahead` with a `count` that is too high. But that would go against removing the panics. I'd gladly have your opinion on this.